### ISSUE TITLE
Lock environment versions

### ISSUE THREAD

**Issue by albinson-arias** [pingbird/puro#33]
I would love to have an option to lock a certain environment's version so i can't upgrade or change it's version without wanting to.

**Comment by pingbird** [comment]
Under what condition would an environment be upgraded without you telling it?

**Comment by albinson-arias** [comment]
I would always have to tell it, let me rephrase, i have a stable env and an env that doesn't need to change it's version never, and i'm always changing between these two, sometimes i want to upgrade my stable env but haven't changed envs (which is completely my fault) and i end up upgrading the env that doesn't never have to be upgraded.

So it would be great to have an option like this or an option to downgrade to last version used in that env.

**Comment by pingbird** [comment]
Are you using `flutter upgrade` instead of `puro upgrade`? The environment argument to `puro upgrade` is non-optional. I understand the muscle memory can cause some issues there :)

**Comment by pingbird** [comment]
Rather than adding a flag to lock an environment I'll add versioned environments which are non upgradable

**Comment by albinson-arias** [comment]
Great, thank you so much!
